src/account_id_ref.rs

```diff
@@ -29,6 +29,33 @@ use crate::{AccountId, ParseAccountError};
 #[cfg_attr(feature = "abi", derive(schemars::JsonSchema, BorshSchema))]
 pub struct AccountIdRef(pub(crate) str);
 
+/// Enum representing possible types of accounts.
+/// This `enum` is returned by the [`get_account_type`] method on [`AccountIdRef`].
+/// See its documentation for more.
+///
+/// [`get_account_type`]: AccountIdRef::get_account_type
+/// [`AccountIdRef`]: struct.AccountIdRef.html
+#[derive(PartialEq)]
+pub enum AccountType {
+    /// Any valid account, that is neither NEAR-implicit nor ETH-implicit.
+    NamedAccount,
+    /// An account with 64 characters long hexadecimal address.
+    NearImplicitAccount,
+    /// An account which address starts with '0x', followed by 40 hex characters.
+    EthImplicitAccount,
+}
+
+impl AccountType {
+    pub fn is_implicit(&self) -> bool {
+        match &self {
+            Self::NearImplicitAccount => true,
+            // TODO(eth-implicit) change to true later, see https://github.com/near/nearcore/issues/10018
+            Self::EthImplicitAccount => false,
```

Comment:
This would look like a breaking change for near-account-id crate, so I would just return `true` here from the get-go. Any strong objections? It would be a separate challenge for the future if NEAR Protocol changes account-id rules drastically.

Reply:
Ok, I can get around that within the nearcore repo, to keep this repo stable.

Reply:
SGTM

Reply:
We still need to work out what to do about the existing named accounts that have the eth account format. I would still go ahead with this PR as agreed because it's blocking everything elsewhere.